File: S2M1/Ruby/bicicleta_carretera.rb
class BicicletaCarretera < Bicicleta 

    def initialize(dorsal=-1, tengoAcabar=false)
        super(dorsal, tengoAcabar)
    end

    def run
        puts "La bicicleta de carretera #{dorsal} ha comenzado la carrera"
        if(acabado)
            sleep(Carrera.abandono_tiempo)
            puts "La bicicleta de carretera #{dorsal} ha abandonado la carrera" 
        else
            sleep(Carrera.DURACION)
            puts "La bicicleta de carretera #{dorsal} ha terminado la carrera"      
        end
    end
end

File: S2M1/Java/src/BicicletaCarretera.java
public class BicicletaCarretera extends Bicicleta {

    BicicletaCarretera(int dorsal, boolean tengoAcabar) {
        super(dorsal, tengoAcabar);
    }

    @Override
    public void run() {
        System.out.println("La bicicleta de carretera " + dorsal + " ha comenzado la carrera");

        try {
            if (acabado){
                Thread.sleep(Carrera.getAbandonoTiempo());
                System.out.println("La bicicleta de carretera " + dorsal + " ha abandonado la carrera");
            }
            else{
                Thread.sleep(Carrera.getDuracion()*1000);
                System.out.println("La bicicleta de carretera " + dorsal + " ha terminado la carrera");
            }
        } catch (InterruptedException e) {
            e.printStackTrace();
        }        

        /*
        if (acabado) {
            System.out.println("La bicicleta de carretera " + dorsal + " ha abandonado la carrera");
        } else {
            System.out.println("La bicicleta de carretera " + dorsal + " ha terminado la carrera");
        }
        */
    }
}